src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -119,6 +119,11 @@ public static void SetArrayItemNodes(ExecutionContext context, ArrayExecutionNod
                     {
                         SetArrayItemNodes(context, arrayNode);
                     }
+                    else if (node is ValueExecutionNode valueNode)
+                    {
+                        node.Result = ((ScalarGraphType)itemType).Serialize(d)
```

Comment:
You check if `node` is `ValueExecutionNode` but then cast different variable `itemType` to `ScalarGraphType`. I propose to change ValueExecutionNode ctor second parameter from `IGraphType` to `ScalarGraphType` and then use here `((ScalarGraphType)valueNode.GraphType).Serialize(d)`.

Reply:
The same suggestions to chenged lines inside `ExecuteNodeAsync`

Reply:
Well, if the resolved value is null, `SetArrayItemNodes` creates a `ValueExecutionNode` with a result of null, no matter if it's a scalar graph type or not.  Would you want to create a `NullExecutionNode` to handle this scenario?

Reply:
Done; added NullExecutionNode as described above.  Please review commit below.

Reply:
Ok